src/test/txpackage_tests.cpp

```diff
@@ -368,6 +545,100 @@ BOOST_FIXTURE_TEST_CASE(package_submission_tests, TestChain100Setup)
+                                                           /*outputs=*/{CTxOut{10*COIN, parent_locking_script}, CTxOut{10*COIN, parent_locking_script},
+                                                                        CTxOut{last_value, parent_locking_script}},
+                                                           /*submit=*/false)};
+        CTransactionRef tx_grandparent = MakeTransactionRef(mtx_grandparent);
+        package_ppfc.push_back(tx_grandparent);
+
+        const CAmount parent_fee{13223};
+        const CAmount parent_value{10*COIN - 13223};
+        auto mtx_parent1{CreateValidMempoolTransaction(/*input_transaction=*/tx_grandparent, /*input_vout=*/0,
+                                                       /*input_height=*/102,
+                                                       /*input_signing_key=*/parent_key,
+                                                       /*output_destination=*/child_locking_script,
+                                                       /*output_amount=*/parent_value, /*submit=*/false)};
+        CTransactionRef tx_parent1 = MakeTransactionRef(mtx_parent1);
+        package_ppfc.push_back(tx_parent1);
+        auto mtx_parent2{CreateValidMempoolTransaction(/*input_transaction=*/tx_grandparent, /*input_vout=*/1,
+                                                       /*input_height=*/102,
+                                                       /*input_signing_key=*/parent_key,
+                                                       /*output_destination=*/child_locking_script,
+                                                       /*output_amount=*/parent_value, /*submit=*/false)};
+        CTransactionRef tx_parent2 = MakeTransactionRef(mtx_parent2);
+        package_ppfc.push_back(tx_parent2);
+
+
+        const CAmount child_fee{2424};
+        const CAmount child_value{last_value + 2*parent_value - child_fee};
+        auto mtx_child{CreateValidMempoolTransaction(/*input_transactions=*/{tx_grandparent, tx_parent1, tx_parent2},
+                                                     /*inputs=*/{COutPoint{tx_grandparent->GetHash(), 2}, COutPoint{tx_parent1->GetHash(), 0}, COutPoint{tx_parent2->GetHash(), 0}},
+                                                     /*input_height=*/102,
+                                                     /*input_signing_keys=*/{parent_key, child_key, grandchild_key},
+                                                     /*outputs=*/{CTxOut{child_value, grandchild_locking_script}},
+                                                     /*submit=*/false)};
+        CTransactionRef tx_child = MakeTransactionRef(mtx_child);
+        package_ppfc.push_back(tx_child);
+
+        // Magic Number Sanity Checks
+        BOOST_CHECK_EQUAL(GetVirtualTransactionSize(*tx_grandparent), 5649);
+        BOOST_CHECK_EQUAL(GetVirtualTransactionSize(*tx_parent1), 191);
+        BOOST_CHECK_EQUAL(GetVirtualTransactionSize(*tx_parent2), 191);
```

Comment:
CI Failure here:

```
C:/Users/ContainerAdministrator/AppData/Local/Temp/cirrus-ci-build/src/test/txpackage_tests.cpp(622): error: in "txpackage_tests/package_submission_tests": check GetVirtualTransactionSize(*tx_parent2) == 191 has failed [190 != 191]
```

I'm going to guess that this is because real signatures are being created, which can occasionally be smaller than 71 bytes, thus resulting in a tx that is one byte smaller. If the script type doesn't matter, I would suggest using taproot keypath spends since that has fixed size signatures. Otherwise these checks will need to account for possibly varying transaction sizes.

Reply:
I'm fine if the magic number checks have a bit of wiggle room tbh, just leave a comment?

Reply:
Changed to use T-T-T-Taproot for fixed size (thanks @achow101)

Reply:
grandparent still not fixed size :facepalm: edited the assertion to be on the multiple of 10. I think it worked this time.

Reply:
taproot coinbase outputs wen

LGTM